### PR TITLE
[MERGE] CS328 Submission : Group 3

### PR DESCRIPTION
**Pull request by Shridhar19110101**
#### Lectures

- `17/01/2022/Monday` : `Clustering, K-center` : `2020_01_17_Clustering.ipynb`
- `18/01/2022/Tuesday` : `k-center, k-means` : `2020_01_17_Clustering.ipynb`

#### Group Seq Number

`3`

#### Assigned Deadline

`29//01/2022/Saturday`

#### Members
- `Shridhar Pawar` : `19110101` : `sominath.ps@iitgn.ac.in` : `Shridhar19110101`
- `Paras Jain` :  `19110096` : `jain.paras@iitgn.ac.in` : `jainparas386`
- `Likhita Baswani` : `19110091` : `likhita.b@iitgn.ac.in` : `likhita29`

### CODE REVIEW

**Comment by yadav-sachin** [comment]
I had added a few comments/suggestions in the PR of Grp 1 and Grp 2.
Check them, many of them will be applicable here as well.

File: CS328-Notes/_toc.yml [hunk 1]
```diff
@@ -13,3 +13,4 @@ parts:
   - file: notebooks/2022_01_03_data_representations.md
   - file: notebooks/2022_01_03_data_representations_notebook.ipynb
   - file: notebooks/2022_01_03_distance_function.md
+  - file: notebooks/2020_01_17_Clustering.ipynb
```
Comment (yadav-sachin) [inline CS328-Notes/_toc.yml]: 2020

File: CS328-Notes/notebooks/2020_01_17_Clustering.ipynb [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+    "\n",
+    "However-   \n",
+    "* How do we define the similarity/distance between the objects?\n",
+    "* When do we call the groups meaningful?\n",
+    "* How many groups should the objects be divided into?\n",
+    "\n",
+    "So typically, there is no supervision for clustering the objects. Since there is no ground truth, evaluating the quality of clustering is often difficult.   \n",
+    "For example, consider the two clustering cases depicted in the image below:\n",
+    "\n",
+    "```{figure} ../assets/2020_01_17_Clustering/fig1.png\n",
```
Comment (yadav-sachin) [inline CS328-Notes/notebooks/2020_01_17_Clustering.ipynb]: Can add a caption to this image. and while referring to this image, can refer to the figure number, just as we do in latex.

This image is taken from an external source? (give credit to the src as well)

File: CS328-Notes/notebooks/2020_01_17_Clustering.ipynb [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+    "\n",
+    "\n",
+    "Cost of points: </b>  \n",
+    "Point 1 = d11    \n",
+    "Point 2 = d12  \n",
+    "Point 3 = d23  \n",
+    "Point 4 = d24  \n",
+    "<b>  \n",
+    "Cost of clusters: </b>   \n",
+    "Cluster 1 = max(d11, d12) = d12  \n",
```
Comment (yadav-sachin) [inline CS328-Notes/notebooks/2020_01_17_Clustering.ipynb]: For mathematical symbols such as d11, d12, D, C1, OPT inside $ $, helps to differentiate from general text.

File: CS328-Notes/notebooks/2020_01_17_Clustering.ipynb [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+    "\n",
+    "\n",
+    "### 2-Approximation Algorithm\n",
+    "Let the input set contain a total of 'n' points.  \n",
+    "Let all the 'k' centers be the points of the input set.\n",
+    "\n",
+    "\n",
+    "Following is the pseudocode for 2-approximation:   \n",
+    "\n",
+    "* Choose centers from given points\n",
```
Comment (yadav-sachin) [inline CS328-Notes/notebooks/2020_01_17_Clustering.ipynb]: https://sphinx-proof.readthedocs.io/en/latest/syntax.html#algorithms
we can use this and write pseudo-code/algorithms more formally.

Comment (yadav-sachin) [reply CS328-Notes/notebooks/2020_01_17_Clustering.ipynb]: 2-approximation, llyod, etc.

File: CS328-Notes/notebooks/2020_01_17_Clustering.ipynb [hunk 5]
```diff
@@ -0,0 +1,493 @@
...
+    "* Choose the first center arbitrarily\n",
+    "* For each remaining center:   \n",
+    "    * Choose the point farthest from the existing centers as the new center   \n",
+    "* Repeat the process until all k centers are chosen\n",
+    "\n",
+    "\n",
+    "<b> Runtime: </b> O(kn)  \n",
```
Comment (yadav-sachin) [inline CS328-Notes/notebooks/2020_01_17_Clustering.ipynb]: $ $

File: CS328-Notes/notebooks/2020_01_17_Clustering.ipynb [hunk 6]
```diff
@@ -0,0 +1,493 @@
...
+    "\n",
+    "Hence    $ \\Delta (i+1)  \\le  \\Delta   (i)$\n",
+    "\n",
+    "\n",
+    "```\n",
+    "\n",
+    "```{prf:lemma}\n",
+    ":label: my-lemma2\n",
+    "\n",
+    "For any two  i, j  where  j < i,    \n",
```
Comment (yadav-sachin) [inline CS328-Notes/notebooks/2020_01_17_Clustering.ipynb]: $i$, $j$

File: CS328-Notes/notebooks/2020_01_17_Clustering.ipynb (hunk 4 above)
Comment (yadav-sachin) [reply CS328-Notes/notebooks/2020_01_17_Clustering.ipynb]: Can we write the algorithms more mathematically? In more formal notations which we generally use pseudo-code.

**Comment by yadav-sachin** [comment]
Few minor changes are required for merge.
1. Delete files ".DS_Store" everywhere in the pull request. There are useless files, I guess.
2. Resolve conflict in Table of Contents file "_toc.yml" (this conflict occurred as changes from other groups were merged.)

Edit: .DS_Store file was due to some other merged commit. I have resolved the final conflicts and merged. 🎊

**Comment by Shridhar19110101** [comment]
It means, I don't have to do anything, right?

**Comment by yadav-sachin** [comment]
@Shridhar19110101 
Right. The submission is complete.

**Pull request merged**